### ISSUE TITLE
golang client library?

### ISSUE THREAD

**Issue by jkanche** [kuzudb/kuzu#2003]
Just wondering if you have any plans to develop a golang client?

**Comment by semihsalihoglu-uw** [comment]
Hi @jkanche, 

Sorry for the late reply. This must have escaped our attention when you posted. The answer is no, we don't yet have plans to develop a go client, primarily because no one in the core team knows go well, so we are hesitant to take the responsibility of maintaining it. We hope that as our community grows, people in community can implement and maintain some of these extensions, as done for example in other open-source systems: [https://github.com/marcboeker/go-duckdb](https://github.com/marcboeker/go-duckdb).

Best,

Semih

**Comment by dezren39** [comment]
Hello do you have any recommendations or guidelines regarding how to build an appropriate kuzu client?

**Comment by mewim** [comment]
Hi @dezren39,

For implmenting a language binding for Go, we suggest you to start with our C API wrapper: https://github.com/kuzudb/kuzu/blob/master/src/include/c_api/kuzu.h and use [cgo](https://pkg.go.dev/cmd/cgo) to create a binding for each function. I think there is not a easy way to directly invoke C++ functions from go, so this seems to be the easist way to get started. Since no one in our team is very familiar with Go, we cannot provide suggestions regarding the API design, etc. of a Go client, but [go-duckdb](https://github.com/marcboeker/go-duckdb) might be a good reference.

**Comment by prrao87** [comment]
@mewim we can close this issue I think?

**Comment by mewim** [comment]
Yes. We now have a Go client at https://github.com/kuzudb/go-kuzu/
